server/events/project_command_runner.go

```diff
@@ -313,11 +314,25 @@ func (p *DefaultProjectCommandRunner) doPlan(ctx models.ProjectCommandContext) (
 		return nil, "", DirNotExistErr{RepoRelDir: ctx.RepoRelDir}
 	}
 
+	lockFileName := ".terraform.lock.hcl"
+	lockfilePath := filepath.Join(projAbsPath, lockFileName)
+	lockFileExistsProirToPlan := common.FileExists(lockfilePath)
+
 	outputs, err := p.runSteps(ctx.Steps, ctx, projAbsPath)
 	if err != nil {
 		if unlockErr := lockAttempt.UnlockFn(); unlockErr != nil {
 			ctx.Log.Err("error unlocking state after plan error: %v", unlockErr)
 		}
+		if !lockFileExistsProirToPlan && common.FileExists(lockfilePath) {
+			// if the lock file did not exist before running the plan delete the lock file that was generated
+			// by terraform init so that on subsequent actions modules and providers are not locked to the versions
+			// detected as a part of this plan
+			ctx.Log.Debug("Deleting `%s` that was generated by terraform init", lockFileName)
+			delErr := os.Remove(lockfilePath)
+			if delErr != nil {
+				ctx.Log.Info("Error Deleting `%s`", lockFileName)
+			}
+		}
```

Comment:
Is there a reason this cleanup logic also can't live in the `InitStepRunner`?

Reply:
I tried that originally but that caused issues with the e2e tests

Reply:
What issues?

Reply:
if `terraform.lock.hcl` does not exist when terraform plan is run it errors with

```shell
│ Error: Could not load plugin
│ 
│ 
│ Plugin reinitialization required. Please run "terraform init".
│ 
│ Plugins are external binaries that Terraform uses to access and manipulate
│ resources. The configuration provided requires plugins which can't be located,
│ don't satisfy the version constraints, or are otherwise incompatible.
│ 
│ Terraform automatically discovers provider requirements from your
│ configuration, including providers used in child modules. To see the
│ requirements and constraints, run "terraform providers".
│ 
│ failed to instantiate provider "registry.terraform.io/hashicorp/null" to obtain schema: unknown provider "registry.terraform.io/hashicorp/null"
│ 
```